stdlib/public/SwiftShims/LibcShims.h

```diff
@@ -60,10 +60,15 @@ __swift_size_t _swift_stdlib_fwrite_stdout(const void *ptr, __swift_size_t size,
                                            __swift_size_t nitems);
 
 // String handling <string.h>
-__attribute__((__pure__)) SWIFT_RUNTIME_STDLIB_INTERFACE __swift_size_t
+#if defined(__clang__) || defined(__GNUC__)
+__attribute__((__pure__))
+#endif
```

Comment:
We usually prefer pulling the attributes themselves into defines, e.g. `SWIFT_RUNTIME_ATTRIBUTE_PURE`.